### ISSUE TITLE
[THIS IS SPAM]           امنا زهرا

### ISSUE THREAD

**Issue by alirezasedmir1991** [exercism/python#4264]
<details>
<summary><mark>TETX OF SPAM MESSAGE</mark></summary>

**Trading Bot AI Project - Comprehensive Roadmap & Documentation**

**Project Title:** AI-Powered Algorithmic Trading Bot

**Repository Purpose:** This repository documents a complete, professional, step-by-step roadmap for building a robust, risk-managed algorithmic trading system with AI capabilities.

---

### Vision & Philosophy
We are not here to gamble. We are building a disciplined, intelligent, data-driven trading system that respects risk, leverages technology, and continuously evolves through learning. Success in trading comes from process, patience, and rigorous validation — not from chasing quick profits.

This project follows a structured, phased approach suitable for individuals or small teams.

---

### Phase 1: Team Formation

**Minimum Required Roles:**
- **Project Manager**: Overall coordination and decision making
- **Python Developer**: Core bot development and integration
- **Financial Markets Specialist**: Strategy design and market knowledge
- **AI Specialist** (Optional): Machine learning models and optimization
- **Capital/Risk Manager**: Position sizing, risk controls, and portfolio management

*Note:* You can start with just a Project Manager + Python Developer and expand later.

---

### Phase 2: Infrastructure Setup

1. **Dedicated Project Email**
   - Create a professional email (e.g., `trading@yourdomain.com` or `ai.trading@yourdomain.com`)
   - Use this email for all project accounts (brokers, APIs, GitHub, etc.)

2. **GitHub Repository**
   - Create a new private/public repository
   - Maintain version control, documentation, and issue tracking

3. **Python Environment**
   ```bash
   python --version  # Should be latest stable (3.10+ recommended)
   ```

4. **Development Tools**
   - Install **VS Code** with extensions: Python, Jupyter, GitHub Copilot (optional)
   - Install **Git**
   - (Optional) Install **Anaconda** for data analysis workflows

---

### Phase 3: Python Mastery Requirements

The developer must be proficient in:
- Variables, Functions, Classes (OOP)
- APIs and JSON handling
- CSV & Data manipulation (pandas, numpy)
- Error Handling & Logging
- Modular code structure

---

### Phase 4: Install Core Libraries

```bash
pip install pandas numpy matplotlib yfinance backtrader vectorbt ta scikit-learn tensorflow
pip install streamlit dash  # For dashboard
```

---

### Phase 5: Financial Markets Fundamentals

Master these concepts before implementing strategies:
- Candlestick patterns
- Trend identification
- Support & Resistance
- Volume analysis
- Technical Indicators: RSI, MACD, EMA, SMA, ATR
- Risk-Reward Ratio
- Position sizing and drawdown management

---

### Phase 6: Data Acquisition

Example using `yfinance`:

```python
import yfinance as yf
import pandas as pd

# Download historical data
data = yf.download("AAPL", period="5y", interval="1d")
data.to_csv("data/AAPL_5y.csv")
```

---

### Phase 7: Strategy Development

**Example Simple Strategy (Crossover):**

```python
def generate_signals(df):
    df['EMA20'] = df['Close'].ewm(span=20).mean()
    df['EMA50'] = df['Close'].ewm(span=50).mean()
    
    df['Signal'] = 0
    df.loc[df['EMA20'] > df['EMA50'], 'Signal'] = 1   # Buy
    df.loc[df['EMA20'] < df['EMA50'], 'Signal'] = -1  # Sell
    return df
```

---

### Phase 8: Backtesting

Evaluate on 5–10 years of data. Key metrics:
- Total Return
- Maximum Drawdown
- Win Rate
- Sharpe Ratio
- Number of Trades
- Profit Factor

Use `backtrader` or `vectorbt` for professional backtesting.

---

### Phase 9: Risk & Money Management (Critical)

**Golden Rule:**
- Risk no more than **1%** of total capital per trade
- Define clear stop-loss and take-profit levels
- Implement position sizing based on ATR or volatility

---

### Phase 10: Web Dashboard

Use **Streamlit** (recommended for speed) or **Dash**.

**Example Streamlit Structure (`dashboard/app.py`):**

```python
import streamlit as st
import pandas as pd
import plotly.express as px

st.title("AI Trading Bot Dashboard")

# Portfolio Overview
col1, col2, col3 = st.columns(3)
col1.metric("Total Balance", "$12,450", "+2.3%")
col2.metric("Today's P&L", "$185", "+1.5%")
col3.metric("Win Rate", "68%", "↑")

# Recent Trades
st.subheader("Recent Transactions")
st.dataframe(pd.read_csv("logs/trades.csv"))

# Performance Charts
fig = px.line(...)  # Equity curve, drawdown, etc.
st.plotly_chart(fig)
```

---

### Phase 11: Paper Trading (Paper Account)

- Run the bot in simulation mode for **minimum 3–6 months**
- Log every decision and outcome
- Only proceed to live trading if consistently profitable with controlled drawdown

---

### Phase 12: Broker Selection & Live Trading

Consider markets:
- US Stocks
- Forex
- Cryptocurrency
- Commodities

**Important for users in Iran:** Carefully research sanctions, use VPNs where necessary, and choose brokers with strong API support.

---

### Phase 13: Broker API Integration

Connect via REST/WebSocket APIs (e.g., Alpaca, Interactive Brokers, Binance, etc.).

Typical flow:
1. Authentication
2. Market data subscription
3. Order placement (Market/Limit/Stop)
4. Position & order status monitoring
5. Error handling & reconnection logic

---

### Phase 14: Live Trading Start

**Begin extremely small** — $50 to $100 maximum for initial live testing.

---

### Phase 15: Monitoring & Continuous Improvement

Daily review questions:
- Why did the system enter/exit this trade?
- Was the loss due to strategy, market regime change, or bug?
- How can we improve?

---

### Phase 16: Integrate Advanced AI

Use AI for:
- Sentiment analysis from news
- Pattern recognition in price/volume
- Hyperparameter optimization
- Regime detection (bull/bear/sideways)

Recommended Tools:
- ChatGPT / Cursor AI / GitHub Copilot → Code assistance
- Google Colab → Heavy ML training
- Notion AI → Documentation

---

### Project Folder Structure

```bash
TradingBot/
├── data/              # Raw and processed market data
├── strategy/          # Strategy logic and indicators
├── ai/                # ML models and optimization
├── backtest/          # Backtesting scripts and results
├── dashboard/         # Streamlit/Dash web app
├── logs/              # Trade logs and errors
├── config/            # API keys, parameters (use .env)
├── tests/             # Unit and integration tests
├── docs/              # Documentation and research
├── utils/             # Helper functions
└── main.py            # Entry point for bot
```

---

### Suggested Timeline (12 Months)

| Month | Focus |
|-------|------|
| 1     | Python mastery + environment setup |
| 2     | Market education + data collection |
| 3     | First strategy implementation |
| 4     | Backtesting & optimization |
| 5     | Dashboard + Paper Trading start |
| 6-8   | Extended Paper Trading & refinement |
| 9     | Small live capital deployment |
| 10-12 | AI integration + continuous improvement |

---

### Costs (Self-Developed)
- Good computer + stable internet
- Time investment (most valuable)
- Possible API/broker fees

---

**Next Steps (Action Items):**
1. Create the GitHub repository and clone this structure.
2. Set up the development environment.
3. Begin Python & market learning simultaneously.

This is a serious professional endeavor. Discipline and patience will determine success.

**"The market is a teacher. Listen carefully, respect risk, and compound knowledge."**

---

I have prepared this as a clean, professional **README.md** ready for your GitHub repository. You can copy it directly.

If you want me to generate the actual Python files (main bot, dashboard, backtester, etc.), specific strategy implementations, or `.gitignore`, `requirements.txt`, and other starter files, just say the word and we will build them step by step with excellence.

Are you ready to initialize the repository and begin Phase 1–2 together? I am here to guide you with full commitment.

</details>

**Comment by BethanyG** [comment]
@alirezasedmir1991,

We consider this malicious spam.  Please do not post to this repo again unless it is directly applicable.
